src/app/repositories/firebase/schema_firebase_repository.py

```diff
@@ -39,7 +77,11 @@ def create_schema(
         schema_metadata (SchemaMetadata): The schema metadata being added to firestore.
         """
 
-        self.schemas_collection.document(schema_id).set(asdict(schema_metadata))
+        transaction.set(
+            firebase_loader.schemas_collection.document(schema_id),
```

Comment:
The `schemas_collection` is already a private field on the schema repository, so you don't need to grab it from the `firebase_loader`, you can just call `self.schemas_collection` now right?

Reply:
Will change. Missed that when I copy my function back from the transaction object